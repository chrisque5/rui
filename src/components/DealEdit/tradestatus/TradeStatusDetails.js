import { useSelector } from 'react-redux';
import { Row, Col } from 'antd';
import {
  getDeal,
} from '../../../utils/selectors';
import { dateFormatter } from '../../../utils/agGrid/formatters';

const TradeStatusDetails = () => {
  const {
    dmsDealReference, dealType, dealAction, dealStatus, tradeTime, executionTime,
  } = useSelector(getDeal);
  const valueMargin = 5;
  const col1LabelWidth = 70;
  const col2LabelWidth = 120;
  return (
    <>
      <Row style={{ marginBottom: 10 }} justify="flex-start">
        <Col>
          <Row data-testid="dmsDealReference">
            <span style={{ width: col1LabelWidth }}><b>DMS Ref: </b></span>
            <span style={{ marginLeft: valueMargin }}>{dmsDealReference}</span>
          </Row>
          <Row data-testid="dealType" style={{ marginTop: valueMargin, marginBottom: valueMargin }}>
            <span style={{ width: col1LabelWidth }}><b>DMS Type: </b></span>
            <span style={{ marginLeft: valueMargin }}>{dealType}</span>
          </Row>
          <Row data-testid="dealAction">
            <span style={{ width: col1LabelWidth }}><b>DMS Action: </b></span>
            <span style={{ marginLeft: valueMargin }}>{dealAction}</span>
          </Row>
        </Col>
        <Col style={{ marginLeft: 200 }}>
          <Row data-testid="tradeDateTime">
            <span style={{ width: col2LabelWidth }}><b>Trade Date/Time: </b></span>
            <span style={{ marginLeft: valueMargin }}>{dateFormatter({ value: tradeTime }, 'DD/MM/YYYY hh:mm:ss')}</span>
          </Row>
          <Row data-testid="executionDateTime" style={{ marginTop: valueMargin, marginBottom: valueMargin }}>
            <span style={{ width: col2LabelWidth }}><b>Execution Date/Time: </b></span>
            <span style={{ marginLeft: valueMargin }}>{dateFormatter({ value: executionTime }, 'DD/MM/YYYY hh:mm:ss')}</span>
          </Row>
          <Row data-testid="dealStatus">
            <span style={{ width: col2LabelWidth }}><b>Deal Status: </b></span>
            <span style={{ marginLeft: valueMargin }}>{dealStatus}</span>
          </Row>
        </Col>
      </Row>
    </>
  );
};

export default TradeStatusDetails;
