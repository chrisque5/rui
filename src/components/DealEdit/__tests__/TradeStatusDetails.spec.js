import {
  renderWithForm, waitFor,
} from 'test-utils/testUtils';
import TradeStatusDetails from '../tradestatus/TradeStatusDetails';
import { deal as mockDeal } from './MockDeal.json';

jest.mock('moment', () => () => jest.requireActual('moment')('2021-07-12'));

jest.mock('../../../utils/selectors.js', () => ({
  ...jest.requireActual('../../../utils/selectors.js'),
  getDeal: jest.fn().mockImplementation(() => mockDeal),
}));

describe('<TradeStatusDetails/>', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders snapshot as expected', () => {
    const { container } = renderWithForm(<TradeStatusDetails />);
    expect(container).toMatchSnapshot();
  });

  test('Verify Deal status details rendered properly', async () => {
    const { getByText, getAllByText } = renderWithForm(<TradeStatusDetails />);
    const dmsRefNo = await waitFor(() => getByText('404'));// 404 is ref no
    expect(dmsRefNo).toBeTruthy();
    const dealType = await waitFor(() => getByText('NDF'));
    expect(dealType).toBeTruthy();
    const dealStatus = await waitFor(() => getByText('VALIDATED'));
    expect(dealStatus).toBeTruthy();
    const dmsAction = await waitFor(() => getByText('AMEND'));
    expect(dmsAction).toBeTruthy();
    const dateFields = await waitFor(() => getAllByText('12/07/2021 12:00:00'));
    expect(dateFields).toBeTruthy();
  });
});
