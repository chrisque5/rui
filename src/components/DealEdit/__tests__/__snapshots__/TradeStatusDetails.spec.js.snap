// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradeStatusDetails/> renders snapshot as expected 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-row-flex-start"
      style="margin-bottom: 10px;"
    >
      <div
        class="ant-col"
      >
        <div
          class="ant-row"
          data-testid="dmsDealReference"
        >
          <span
            style="width: 70px;"
          >
            <b>
              DMS Ref: 
            </b>
          </span>
          <span
            style="margin-left: 5px;"
          >
            404
          </span>
        </div>
        <div
          class="ant-row"
          data-testid="dealType"
          style="margin-top: 5px; margin-bottom: 5px;"
        >
          <span
            style="width: 70px;"
          >
            <b>
              DMS Type: 
            </b>
          </span>
          <span
            style="margin-left: 5px;"
          >
            NDF
          </span>
        </div>
        <div
          class="ant-row"
          data-testid="dealAction"
        >
          <span
            style="width: 70px;"
          >
            <b>
              DMS Action: 
            </b>
          </span>
          <span
            style="margin-left: 5px;"
          >
            AMEND
          </span>
        </div>
      </div>
      <div
        class="ant-col"
        style="margin-left: 200px;"
      >
        <div
          class="ant-row"
          data-testid="tradeDateTime"
        >
          <span
            style="width: 120px;"
          >
            <b>
              Trade Date/Time: 
            </b>
          </span>
          <span
            style="margin-left: 5px;"
          >
            12/07/2021 12:00:00
          </span>
        </div>
        <div
          class="ant-row"
          data-testid="executionDateTime"
          style="margin-top: 5px; margin-bottom: 5px;"
        >
          <span
            style="width: 120px;"
          >
            <b>
              Execution Date/Time: 
            </b>
          </span>
          <span
            style="margin-left: 5px;"
          >
            12/07/2021 12:00:00
          </span>
        </div>
        <div
          class="ant-row"
          data-testid="dealStatus"
        >
          <span
            style="width: 120px;"
          >
            <b>
              Deal Status: 
            </b>
          </span>
          <span
            style="margin-left: 5px;"
          >
            VALIDATED
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
